elastic/apm-server
# Beat: Allow setting beat name in configuration

## Motivation/summary

Uses the beat name if set in the configuration file. Restores what used to work on 7.x and perhaps prior to #9149 being introduced.

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Set `name: my-apm-server` under the APM config in `testing/infra/k8s/base/stack/apm-server.yaml`
1. Tilt up
1. Navigate to Stack Monitoring -> Verify that APM Server instance is named `my-apm-server`.
